### PR TITLE
Add poetry to VALID_PYTHON_BUILD_BACKENDS

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
* [x] Added a ``news`` entry
* [x] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

### CODE REVIEW

**Comment by beckermr** [comment]
To make a news entry, copy the template file to a new file in the `news` directory and then fill it out.

**Comment by beckermr** [comment]
Doesn't matter but putting the pr number in the file name is helpful.

**Comment by thewchan** [comment]
@beckermr done, thanks!

File: news/pr-2175.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+**Changed:**
+
+* Add Poetry as a valid build backend
```
Comment (beckermr) [inline news/pr-2175.rst:3]:
```suggestion
* Added ``poetry`` as a valid ``pip`` build backend. (#2175)
```

**Pull request merged**